lib/Actions.js

```diff
@@ -115,7 +172,46 @@ Actions.prototype = {
         }
 
         return vow.all([actionPromise, waitingForPage]).spread(function (result) {
-            return result;
+            return casesPromise || result;
+        });
+    },
+
+    _performCases: function (cases, parentSelector) {
+        debug('handle several cases in parallel %o', cases);
+
+        var wonCase = null;
+        var promises = cases.map(function (actions, caseNumber) {
+            var beginningPromise = this._performAction(actions[0], parentSelector);
```

Comment:
I understood the profit of it, just care about calling actions[0] without any default value. Could be empty